src/dmd/dsymbolsem.d

```diff
@@ -3480,6 +3480,27 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
             }
         }
 
+        if (sc.flags & (SCOPE.printf | SCOPE.scanf))
+        {
+            /* printf/scanf-like functions must be of the form:
+             *    extern (C) T printf([parameters...], const(char)* format, ...);
+             * or:
+             *    extern (C) T vprintf([parameters...], const(char)* format, va_list);
+             */
+            if (f.linkage != LINK.c ||
```

Comment:
Why limit it to `extern(C)` though ?

Reply:
I agree, this could be extended to any function that either wraps around, or performs an action that is compatible with  scanf/printf.

Reply:
I'll add `extern (C++)`. There's no need for `extern (D)` or the others.

Reply:
Wrapper functions can be `extern(D)`, see e.g. some of the `error`, `warning` and `deprecation` functions in dmd.